internal/generate/clusterserviceversion/clusterserviceversion_test.go

```diff
@@ -0,0 +1,379 @@
+		operatorType = projutil.OperatorTypeGo
+		version      = "0.0.1"
+	)
+
+	BeforeEach(func() {
+		buf = &bytes.Buffer{}
+	})
+
+	Describe("for the new Go project layout", func() {
+
+		Context("with correct Options", func() {
+
+			var (
+				tmp string
+				err error
+			)
+
+			BeforeEach(func() {
+				tmp, err = ioutil.TempDir(".", "")
+				ExpectWithOffset(1, err).ToNot(HaveOccurred())
+			})
+
+			AfterEach(func() {
+				if tmp != "" {
+					os.RemoveAll(tmp)
+				}
+			})
+
+			It("should write a ClusterServiceVersion manifest to an io.Writer", func() {
+				g = Generator{
+					OperatorName: operatorName,
+					OperatorType: operatorType,
+					Version:      version,
+					Collector:    col,
+				}
+				opts := []Option{
+					WithBase(csvBasesDir, goAPIsDir),
+					WithWriter(buf),
+				}
+				Expect(g.Generate(cfg, opts...)).ToNot(HaveOccurred())
+				Expect(buf.String()).To(MatchYAML(newCSVStr))
+			})
+			It("should write a ClusterServiceVersion manifest to a base file", func() {
+				g = Generator{
+					OperatorName: operatorName,
+					OperatorType: operatorType,
+				}
+				opts := []Option{
+					WithBase(csvBasesDir, goAPIsDir),
+					WithBaseWriter(tmp),
+				}
+				Expect(g.Generate(cfg, opts...)).ToNot(HaveOccurred())
+				outputFile := filepath.Join(tmp, "bases", makeCSVFileName("memcached-operator"))
+				Expect(outputFile).To(BeAnExistingFile())
+				Expect(string(readFileHelper(outputFile))).To(MatchYAML(baseCSVUIMetaStr))
+			})
+			It("should write a ClusterServiceVersion manifest to a bundle file", func() {
+				g = Generator{
+					OperatorName: operatorName,
+					OperatorType: operatorType,
+					Version:      version,
+					Collector:    col,
+				}
+				opts := []Option{
+					WithBase(csvBasesDir, goAPIsDir),
+					WithBundleWriter(tmp),
+				}
+				Expect(g.Generate(cfg, opts...)).ToNot(HaveOccurred())
+				outputFile := filepath.Join(tmp, bundle.ManifestsDir, makeCSVFileName("memcached-operator"))
+				Expect(outputFile).To(BeAnExistingFile())
+				Expect(string(readFileHelper(outputFile))).To(MatchYAML(newCSVStr))
+			})
+		})
+
+		Context("with incorrect Options", func() {
+
+			BeforeEach(func() {
+				g = Generator{
+					OperatorName: operatorName,
+					OperatorType: operatorType,
+					Version:      version,
+					Collector:    col,
+				}
+			})
+
+			It("should return an error without any Options", func() {
+				opts := []Option{}
+				Expect(g.Generate(cfg, opts...)).To(MatchError(genutil.InternalError("getWriter must be set")))
+			})
+			It("should return an error without a getWriter", func() {
+				opts := []Option{
+					WithBase(csvBasesDir, goAPIsDir),
+				}
+				Expect(g.Generate(cfg, opts...)).To(MatchError(genutil.InternalError("getWriter must be set")))
+			})
+			It("should return an error without a getBase", func() {
+				opts := []Option{
+					WithWriter(&bytes.Buffer{}),
+				}
+				Expect(g.Generate(cfg, opts...)).To(MatchError(genutil.InternalError("getBase must be set")))
```

Comment:
should we not check for it case if the files expected was or not gen as well?

Reply:
In a follow-up.